Assets/Scripts/GameTime/Clock.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Managers;
+using UnityEngine;
+
+namespace GameTime
+{
+    public class Clock : MonoBehaviour
+    {
+        [SerializeField][Tooltip("Day length in minutes")]
+        private float dayLength;
+    
+        [SerializeField][Tooltip("Year length in days")]
+        private int yearLength;
+
+        [SerializeField] [Tooltip("Current time")] [Range(0f, 1f)]
+        private float timeOfDay;
+        public float TimeOfDay => timeOfDay;
+
+        private int _days;
+        public int Days => _days;
+
+        private int _years;
+        public int Years => _years;
+
+        private float _timeScale;
+        
+        [SerializeField][Tooltip("Controls timeScale so certain time periods (night) can pass faster than others (day)")]
+        private AnimationCurve timeCurve;
+
+        private float _timeCurveNormalization;
+
+        private void Awake()
+        {
+            NormalizeTimeCurve();
+        }
+
+        private void Update()
+        {
+            UpdateTimeScale();
+            UpdateTime();
+            _timeScale = 24 / (dayLength / 60); // calculating basic tick rate
```

Comment:
you moved it from UpdateTimeScale to Update? 😀

Reply:
Could you please move this to Awake or to the field declaration @tobihoffmann? :)

Reply:
oh wow. :D